### ISSUE TITLE
Address details not visible when clicking on details &  ignoring OSM addr: tags 

### ISSUE THREAD

**Issue by duiveldoder** [osmandapp/OsmAnd#17066]
Description
When I select a building - it retrieves the `addr:housenumber` from that building.

However, the `addr:street` it doesn't!
Instead - even if this data is available - OSMAnd ignores the physical reality on the ground and searches around the area for any street it finds .. even if that's totally utterly nonsense..

I'll add a screenshot to house 63 A, which is the same for 63, 63 B all show the same nonsense data..

Furthermore - also addr:village, addr:city, addr:postcode, addr:municipality, addr:province, addr:country, and all other legit addr:.... tags are ignored by OSMAnd.

<table style="width:100%">
  <tr>
    <th>This below is the screenshot of the actual data for the house on OSM - screencapture from ID Editor
</th>
    <th>Below is the false information OSMAnd gives me:
</th>
  </tr>
  <tr>
    <td><img src="https://user-images.githubusercontent.com/15773876/234710601-a41a9ff6-1d44-4a2d-91e4-22bcbb7964e0.png"  width="300" height="auto"/> </td>
    <td><img src="https://user-images.githubusercontent.com/15773876/234710806-520adf6c-2b7a-412c-811b-2e912500d060.png"  width="300" height="auto"/> </td>
  </tr>
</table>   

### Steps to reproduce
Well just click on any building with addr: details

### Actual result
Correct `addr:housenumber`, but all other data is being guessed - which is okay I guess if that data were missing on the object.
But if the object already has the correct data anyway - don't look around for possibly false data

### Expected result
the correct address? LOL

### Your Environment
OsmAnd Version: latest nightly always
Android/iOS version: 10
Device model: Huawei 20 Mate Pro

### SOLUTION ###
Have OSMANd to show the actual addr:..... if one clicks on details..

Then I wish to see the same - as what I see in ID Editor

### Note ### 
This is very important - as right now - 63 at Cordillera Road is actually the COMPLETE WRONG ADDRESS!
And I noticed during my time here - many occasions just like this - where you end up with a complete wrong address.
Because OSMAnd ignores the physical reality on the ground and "magically casts it's own physical reality by guessing"

**Comment by duiveldoder** [comment]
Example for 63

Physical reality on the ground:
![image](https://user-images.githubusercontent.com/15773876/234711784-b966c7ec-a948-49bf-85f0-414c89cfdef7.png)

What OSMAnd magically casts out of the belly:
![image](https://user-images.githubusercontent.com/15773876/234712071-59e004e0-69a7-445a-9d1e-a9695c7b2ab4.png)

But 63 Cordillera Road is actually:
![image](https://user-images.githubusercontent.com/15773876/234712274-9405b711-6c7e-4c0e-899d-c12222432949.png)

The first mentioned is not Cordillera Road but place=Purok 20 and neighbourhood=Police Village

**Comment by duiveldoder** [comment]
Another example:
![image](https://user-images.githubusercontent.com/15773876/234712658-ae4d0f65-a57e-4425-b55c-4db1262f02fa.png)

The actual physical reality is however:
![image](https://user-images.githubusercontent.com/15773876/234712768-89dbf014-5548-4f30-9b6c-4ef4af49c594.png)

**Comment by duiveldoder** [comment]
Even MUCH WORSE for POI!

Here an example:
![image](https://user-images.githubusercontent.com/15773876/234713296-e4763bf7-c70f-458c-b5f1-6318dfe9bad1.png)

That location is part of the house 63... and it's neither in Purok 22, nor at Cordillera road nor house 62 A..

Here is the actual physical reality on the ground which OSMAnd totally ignores:
![image](https://user-images.githubusercontent.com/15773876/234713568-c09c1fc9-ed0a-46dc-a314-8b29efdab333.png)

As one can see - the data on the OSM Database is 100% accurate - but OSMAnd claims it knows it all much better than the OSM database and knows instead of using the actual physical reality on the ground - it's better to just randomly guess some utter nonsense..

Sorry to be so direct but this has to be fixed - this issue is open for 2+ years now - it was reported before already years ago - and still not fixed..

please use the addr: keys where they exist..

and show them in the details..

and resort to guessing ONLY if there's no proper addr: keys to the objects/nodes/ways/POI, etc..

**Comment by Zirochkabila** [comment]
OsmAnd~ 4.5.0#35865m, Released: 2023-04-29
Can be reproduced using Live updates
`16.41037° N, 120.59013° E`
[OSM](https://www.openstreetmap.org/way/1164984225#map=19/16.41042/120.59077)

<table style="width:100%">
  <tr>
<th><img src="https://user-images.githubusercontent.com/104760013/235347592-366b26c6-3feb-46a7-ae5b-fa1b92d3649b.jpeg"  width="300" height="auto"/> </th>
 <th><img src="https://user-images.githubusercontent.com/104760013/235347609-eb0aeeef-7c08-4d01-a185-46f0be06bcfc.png"  width="600" height="auto"/> </th>
  </tr>
</table>

**Comment by Zirochkabila** [comment]
Duplicate https://github.com/osmandapp/OsmAnd/issues/10085

**Comment by vshcherb** [comment]
we don't store in POI addr information and run geocoding request based on the point, so in that case we aim to provide better geocoding algorithm

**Comment by duiveldoder** [comment]
It's not a duplicated issue really.

The other tickets are about the search functionality.

I didn't report on the search functionality - I reported on clicking on a specific location/building and getting wrong data about that building/location.

That's not the same as a wrong result from a search query.

**Comment by vshcherb** [comment]
This is not the duplicate of mentionned issues clearly its' a mistake but it is a known issue that we don't use object information addr: tags cause we try to use reverse geocoding approach and similar to Nominatim to use admin polygons to build structure however we don't build information about provinces / etc... (only City (District) / Street / House).

We don't want to save extra address information in the points cause it wll make maps larger and doesn't improve our reverse geocoding algorithms

**Comment by duiveldoder** [comment]
> This is not the duplicate of mentionned issues clearly its' a mistake but it is a known issue that we don't use object information addr: tags cause we try to use reverse geocoding approach and similar to Nominatim to use admin polygons to build structure however we don't build information about provinces / etc... (only City (District) / Street / House).
> 
> We don't want to save extra address information in the points cause it wll make maps larger and doesn't improve our reverse geocoding algorithms

From the "we don't want" i understand it's per design and desire that users get completely wrong information? Did I understand that correct?

**Comment by vshcherb** [comment]
Information should be correct, implementation could be different. You imply that we should take addr: details from osm object, we think implementation should be different end result should be the same.

**Comment by vshcherb** [comment]
As I tested only https://www.openstreetmap.org/way/762572645 this building is supported properly, cause it has addr:street, other have quite strange different mapping:
[addr:neighbourhood](https://wiki.openstreetmap.org/wiki/Key:addr:neighbourhood?uselang=en)
[addr:place](https://wiki.openstreetmap.org/wiki/Key:addr:place?uselang=en)

It looks like we first of all need to fix the search issue and only after that detecting address

**Comment by duiveldoder** [comment]
addr:place is totally valid according to the wiki

It's used in all places where there is NO STREET NAME

Then addr:place is the substitute for the fact there's no street name.

**Comment by duiveldoder** [comment]
Why did you close this? That's very rude and arrogant. I'm a user asking for help to help OSMand getting better and it's just being closed aka ignored.

If OSMAnd as you say doesn't support the approved tag of addr:place - then instead of ignoring approved tagging schemes - OSMAnd could also support approved tagging schemes?

**Comment by goldfndr** [comment]
#4899 is closely related

**Comment by duiveldoder** [comment]
> #4899 is closely related

I beg to differ, it's not even closely related!

My issue addresses the fact that addr:place - which is formally adopted by the OSM community as an alternative to addr:street - is completely lacking in OSMAnd.

**Comment by goldfndr** [comment]
It's closely related to the original post, which stated that OsmAnd ignores addr:* data tagged directly on the POI and searches around the area for addresss data.

**Comment by duiveldoder** [comment]
> It's closely related to the original post, which stated that OsmAnd ignores addr:* data tagged directly on the POI and searches around the area for addresss data.

again - I beg to differ. Especially your argumentation - if the symptoms are similar - then it's close enough

every single professional educated engineer, doctor, etc in the world will have a good laugh at that statement.

Just because some symptoms are similar, or even identical - doesn't mean the problem itself or the disease or bug we're dealing with is related.

**Comment by vshcherb** [comment]
> Why did you close this? That's very rude and arrogant. I'm a user asking for help to help OSMand getting better and it's just being closed aka ignored.

I think you don't try to understand what should be fixed, what won't be fixed, what's the strategy. Nothing personal to you at all though you are not helping to find the root cause.

1. 62 A str. Cordillera Road (Purok 22) city Baguio dist=1 - is doing reverse geocoding fine cause it has addr:street (https://www.openstreetmap.org/way/762572645#map=19/16.41387/120.56339)
2. without addr:street reverse geocodind won't work and it's a duplicate already for an issue https://github.com/osmandapp/OsmAnd/issues/16127

So there is no need to create other issues.

@duiveldoder : If you continue to discuss people and their actions, you will be banned.

**Comment by duiveldoder** [comment]
> @duiveldoder : If you continue to discuss people and their actions, you will be banned.

Would you please ban me from OSMAnd on Github then?

Thank you
